src/main/java/org/pesmypetcare/webservice/entity/medalmanager/UserMedalEntity.java

```diff
@@ -0,0 +1,67 @@
+package org.pesmypetcare.webservice.entity.medalmanager;
+
+import lombok.Data;
+
+import java.util.List;
+
+/**
+ * @author Oriol Catalán
+ */
+@Data
+public class UserMedalEntity {
+    public static final String NAME = "name";
+    public static final String PROGRESS = "progress";
+    public static final String CURRENT_LEVEL = "currentLevel";
+    public static final String COMPLETED_LEVELS_DATE = "completedLevelsDate";
+    private String name;
+    private Double progress;
+    private Double currentLevel;
+    private List<String> completedLevelsDate;
+
+    public UserMedalEntity() { }
+
+    public UserMedalEntity(String name, Double progress, Double currentLevel, List<String> completedLevelsDate) {
+        for (String date : completedLevelsDate) {
+            checkDateFormat(date);
+        }
+        this.name = name;
+        this.progress = progress;
+        this.currentLevel = currentLevel;
+        this.completedLevelsDate = completedLevelsDate;
+    }
+
+    /**
+     * Checks that field has the correct format for a Medal simple attribute.
```

Comment:
Remove simple here, simply do a CTRL+F in all your files and look for all simples and if they shouldn't be there remove them